lib/zendesk_api/collection.rb

```diff
@@ -356,30 +356,27 @@ def intentional_obp_request?
       Helpers.present?(@options["page"]) && !cbp_request?
     end
 
-    def should_try_cbp?(path_query_link)
-      not_supported_endpoints = %w[show_many]
-      not_supported_endpoints.none? { |endpoint| path_query_link.end_with?(endpoint) }
+    def supports_cursor_based_pagination?
+      @resource_class.const_defined?(:CBP_ACTIONS) && @resource_class.const_get(:CBP_ACTIONS).any? { |supported_path| path.end_with?(supported_path) }
+    end
+
+    def first_of_multiple_pages?
+      @next_page.nil?
+    end
+
+    def modify_request_for_cbp
```

Comment:
This is good.

Minor: as a private method, I wonder if `set_cbp_options` would be more explicit.